cep-9999.md

```diff
@@ -0,0 +1,80 @@
+<tr><td> Implementation </td><td> N/A </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize names and other strings used throughout the conda ecosystem.
+
+## Specification
+
+### Artifacts
+
+The conda ecosystem distinguishes between two types of packages:
+
+- Regular package names: represented by a concrete, downloadable, extractable conda artifact.
+- Virtual package names: not backed by any physical artifact. They only exist on the client side.
+
+### Package names
+
+A regular package name MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. It MUST start with a letter, a number, or a single underscore. It MUST NOT start with two consecutive underscores.
+
+Virtual package names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST start with two underscores.
+
+This means that regular package names are limited to those which match the following case-insensitive regex: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+Virtual package names MUST follow this other regex: `^__([a-z0-9]+(\.|-|_|$))+$`.
+
+### Version strings
+
+Version strings MUST only consist of digits, periods, lowercase ASCII letters, underscores, hyphens, plus signs and exclamation marks. Hyphens and underscores MAY be used interchangeably, but they MUST NOT be mixed in the same version string.
+
+Version strings MAY specify an epoch field, which MUST be an integer separated by a single exclamation mark from the rest of the version information. If not specified, the epoch is considered to be `0`. The exclamation mark MUST NOT be used anywhere else.
+
+The version string SHOULD start with a digit.
+
+The plus sign SHOULD separate the regular version information from the local version information.
+The plus sign SHOULD NOT be used for any other purposes.
+
+Multiple separator characters (periods, underscores, plus signs) SHOULD NOT be used contiguously.
+
+The following regex does not cover all rules, but all valid version strings MUST match it: `^([0-9]+!)?[0-9][0-9a-z.(_|-)+]*$`.
+
+### Build strings
+
+Builds strings MUST only consist of lowercase ASCII letters, numbers, and underscores. They MUST match this regex `^[a-zA-Z0-9_]+$`.
+
+### Filenames
+
+Regular conda artifacts MUST be named following this scheme:
+
+```
+<package name>-<version string>-<build string>.<extension>
+```
+
+Virtual conda packages SHOULD NOT need filename standardization.
+
+### Channels
+
+#### Channel names
+Channel names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST start and end with a letter or a digit. In other words, they MUST match  this regex:  `^[a-z0-9]+((-|_|.)[a-z0-9]+)*$`.
+
+#### Label names
+
+Channel label names MUST only consist of ASCII letters, digits, underscores, hyphens, slashes, periods, colons and whitespace. They MUST start with a letter. They MUST match this regex: `^[a-zA-Z][0-9a-zA-Z_\-\.\/:\s]*`.
```

Comment:
The reserved `NOLABEL` label for the label empty set needs to be specified here.

Reply:
Well the issue is how compatible across registries we want things to be. If people are using NOLABEL as an actual label on anaconda.org, then that package could not be mirrored to an OCI registry. So I think we'd ideally want it everywhere.